app/src/main/java/com/google/android/stardroid/views/PreferencesButton.java

```diff
@@ -103,6 +103,38 @@ private void setPreference() {
     Log.d(TAG, "Setting preference " + prefKey + " to... " + isOn);
     if (prefKey != null) {
       preferences.edit().putBoolean(prefKey, isOn).apply();
+      String s;
+      if(prefKey.equals("source_provider.0") && isOn){
```

Comment:
So I see your good intention here, but I'm not sure this is the best way to achieve it. Honestly, this UI is very old and needs a refresh. I'm far from convinced that these filters need to be on the main screen all the time like this since I doubt most users interact with that all that often. So in any refresh I think this functionality would be handled differently.

In any case, just looking at the change itself, the problem is I think that your strings won't be localized if you hard code them like this, and Sky Map is used all over the world. (Admittedly our translations have gaps, but I don't want to make them worse).

Bug fixes are always welcome, but can I respectfully suggest that when you see a change to functionality you want to make, you check in first to see if that's a direction the devs want to go?  I will admit that we're pretty slow to respond on bug/feature request threads these days, for which I apologise.

Reply:
okay, got it. Next time I'll discuss it before sending a PR and look for the changes that are more user-friendly. Thanks for your review and suggestions.